instance/models/server.py

```diff
@@ -256,7 +257,59 @@ def update_status(self):
         raise NotImplementedError
 
 
-class OpenStackServer(Server):
+class OpenStackServerConfiguration(models.Model):
+    """
+    Configuration fields used by OpenStack Servers
+
+    Mutable on the instance but immutable on the Server/AppServer.
+    """
+    class Meta:
+        abstract = True
+
+    # openstack server settings
+    openstack_server_flavor = JSONField(
+        null=True, blank=True, default=None,
+        help_text='JSON openstack flavor selector, e.g. {"name": "vps-ssd-1"}.'
+                  ' Defaults to settings.OPENSTACK_SANDBOX_FLAVOR on server creation.',
+    )
+    openstack_server_base_image = JSONField(
+        null=True, blank=True, default=None,
+        help_text='JSON openstack base image selector, e.g. {"name": "ubuntu-12.04-ref-ul"}'
+                  ' Defaults to settings.OPENSTACK_SANDBOX_BASE_IMAGE on server creation.',
+    )
+    openstack_server_ssh_keyname = models.CharField(
+        max_length=256,
+        null=True, blank=True, default='',
+        help_text='SSH key name used when setting up access to the openstack project.'
+                  ' Defaults to settings.OPENSTACK_SANDBOX_SSH_KEYNAME on server creation.',
+    )
```

Comment:
@pomegranited Since these settings are only set once per VM/AppServer, and can then be retrieved from the Nova API, is it useful to store it at the individual AppServer model level, vs setting it at the instance level and then using the current instance value when provisioning a new AppServer?